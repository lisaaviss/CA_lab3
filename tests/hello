{
    "code": [
        {
            "opcode": "out",
            "arg2": "104",
            "arg2_type": "const"
        },
        {
            "opcode": "out",
            "arg2": "101",
            "arg2_type": "const"
        },
        {
            "opcode": "out",
            "arg2": "108",
            "arg2_type": "const"
        },
        {
            "opcode": "out",
            "arg2": "108",
            "arg2_type": "const"
        },
        {
            "opcode": "out",
            "arg2": "111",
            "arg2_type": "const"
        },
        {
            "opcode": "out",
            "arg2": "32",
            "arg2_type": "const"
        },
        {
            "opcode": "out",
            "arg2": "119",
            "arg2_type": "const"
        },
        {
            "opcode": "out",
            "arg2": "111",
            "arg2_type": "const"
        },
        {
            "opcode": "out",
            "arg2": "114",
            "arg2_type": "const"
        },
        {
            "opcode": "out",
            "arg2": "108",
            "arg2_type": "const"
        },
        {
            "opcode": "out",
            "arg2": "100",
            "arg2_type": "const"
        },
        {
            "opcode": "halt"
        }
    ],
    "data": [
        "0"
    ]
}