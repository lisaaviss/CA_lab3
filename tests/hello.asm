section text
out 'h'
out 'e'
out 'l'
out 'l'
out 'o'
out ' '
out 'w'
out 'o'
out 'r'
out 'l'
out 'd'