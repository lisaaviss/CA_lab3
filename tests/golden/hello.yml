source: |-
  let temp = "";
  input(temp);
  while (temp != EOF)
  {
  print(temp);
  input(temp);
  }

input: |-
  privet

code: |-
  [
      {
          "opcode": "ld",
          "arg1": "rx3",
          "arg2": 0
      },
      {
          "opcode": "wr",
          "arg1": "rx3"
      },
      {
          "opcode": "ld",
          "arg1": "rx2",
          "arg2": 0
      },
      {
          "opcode": "input"
      },
      {
          "opcode": "ld",
          "arg1": "rx15",
          "arg2": 15
      },
      {
          "opcode": "ld",
          "arg1": "rx2",
          "arg2": 0
      },
      {
          "opcode": "ld",
          "arg1": "rx4",
          "arg2": "rx2"
      },
      {
          "opcode": "je",
          "arg1": "rx4",
          "arg2": "rx0"
      },
      {
          "opcode": "ld",
          "arg1": "rx2",
          "arg2": 0
      },
      {
          "opcode": "ld",
          "arg1": "rx5",
          "arg2": "rx2"
      },
      {
          "opcode": "print",
          "arg1": "rx5",
          "arg2": 1
      },
      {
          "opcode": "ld",
          "arg1": "rx2",
          "arg2": 0
      },
      {
          "opcode": "input"
      },
      {
          "opcode": "ld",
          "arg1": "rx15",
          "arg2": 4
      },
      {
          "opcode": "jmp"
      },
      {
          "opcode": "halt"
      }
  ]

log: |
  DEBUG    root:machine.py:335 {TICK: 4, RX1: 1, RX2: 0, RX3: 0, RX4: 0, RX5: 0, RX6: 0, RX7: 0, RX8: 0, RX9: 0, RX10: 0, RX11: 0, RX12: 2047, RX13: 0, RX14: 0, RX15: 0}
  DEBUG    root:machine.py:335 {TICK: 8, RX1: 2, RX2: 1, RX3: 0, RX4: 0, RX5: 0, RX6: 0, RX7: 0, RX8: 0, RX9: 0, RX10: 0, RX11: 0, RX12: 2047, RX13: 0, RX14: 0, RX15: 0}
  DEBUG    root:machine.py:335 {TICK: 12, RX1: 3, RX2: 0, RX3: 0, RX4: 0, RX5: 0, RX6: 0, RX7: 0, RX8: 0, RX9: 0, RX10: 0, RX11: 0, RX12: 2047, RX13: 0, RX14: 0, RX15: 0}
  DEBUG    root:machine.py:335 {TICK: 16, RX1: 4, RX2: 1, RX3: 0, RX4: 0, RX5: 0, RX6: 0, RX7: 0, RX8: 0, RX9: 0, RX10: 0, RX11: 0, RX12: 2047, RX13: 0, RX14: 0, RX15: 0}
  DEBUG    root:machine.py:335 {TICK: 20, RX1: 5, RX2: 1, RX3: 0, RX4: 0, RX5: 0, RX6: 0, RX7: 0, RX8: 0, RX9: 0, RX10: 0, RX11: 0, RX12: 2047, RX13: 0, RX14: 0, RX15: 15}
  DEBUG    root:machine.py:335 {TICK: 24, RX1: 6, RX2: 0, RX3: 0, RX4: 0, RX5: 0, RX6: 0, RX7: 0, RX8: 0, RX9: 0, RX10: 0, RX11: 0, RX12: 2047, RX13: 0, RX14: 0, RX15: 15}
  DEBUG    root:machine.py:335 {TICK: 28, RX1: 7, RX2: 0, RX3: 0, RX4: 112, RX5: 0, RX6: 0, RX7: 0, RX8: 0, RX9: 0, RX10: 0, RX11: 0, RX12: 2047, RX13: 0, RX14: 0, RX15: 15}
  DEBUG    root:machine.py:335 {TICK: 32, RX1: 8, RX2: 0, RX3: 0, RX4: 112, RX5: 0, RX6: 0, RX7: 0, RX8: 0, RX9: 0, RX10: 0, RX11: 0, RX12: 2047, RX13: 0, RX14: 0, RX15: 15}
  DEBUG    root:machine.py:335 {TICK: 36, RX1: 9, RX2: 0, RX3: 0, RX4: 112, RX5: 0, RX6: 0, RX7: 0, RX8: 0, RX9: 0, RX10: 0, RX11: 0, RX12: 2047, RX13: 0, RX14: 0, RX15: 15}
  DEBUG    root:machine.py:335 {TICK: 40, RX1: 10, RX2: 0, RX3: 0, RX4: 112, RX5: 112, RX6: 0, RX7: 0, RX8: 0, RX9: 0, RX10: 0, RX11: 0, RX12: 2047, RX13: 0, RX14: 0, RX15: 15}
  INFO     root:machine.py:57 output: [] << 'p'
  DEBUG    root:machine.py:335 {TICK: 43, RX1: 11, RX2: 0, RX3: 0, RX4: 112, RX5: 112, RX6: 0, RX7: 0, RX8: 0, RX9: 0, RX10: 0, RX11: 0, RX12: 2047, RX13: 0, RX14: 0, RX15: 15}
  DEBUG    root:machine.py:335 {TICK: 47, RX1: 12, RX2: 0, RX3: 0, RX4: 112, RX5: 112, RX6: 0, RX7: 0, RX8: 0, RX9: 0, RX10: 0, RX11: 0, RX12: 2047, RX13: 0, RX14: 0, RX15: 15}
  DEBUG    root:machine.py:335 {TICK: 51, RX1: 13, RX2: 1, RX3: 0, RX4: 112, RX5: 112, RX6: 0, RX7: 0, RX8: 0, RX9: 0, RX10: 0, RX11: 0, RX12: 2047, RX13: 0, RX14: 0, RX15: 15}
  DEBUG    root:machine.py:335 {TICK: 55, RX1: 14, RX2: 1, RX3: 0, RX4: 112, RX5: 112, RX6: 0, RX7: 0, RX8: 0, RX9: 0, RX10: 0, RX11: 0, RX12: 2047, RX13: 0, RX14: 0, RX15: 4}
  DEBUG    root:machine.py:335 {TICK: 57, RX1: 4, RX2: 1, RX3: 0, RX4: 112, RX5: 112, RX6: 0, RX7: 0, RX8: 0, RX9: 0, RX10: 0, RX11: 0, RX12: 2047, RX13: 0, RX14: 0, RX15: 4}
  DEBUG    root:machine.py:335 {TICK: 61, RX1: 5, RX2: 1, RX3: 0, RX4: 112, RX5: 112, RX6: 0, RX7: 0, RX8: 0, RX9: 0, RX10: 0, RX11: 0, RX12: 2047, RX13: 0, RX14: 0, RX15: 15}
  DEBUG    root:machine.py:335 {TICK: 65, RX1: 6, RX2: 0, RX3: 0, RX4: 112, RX5: 112, RX6: 0, RX7: 0, RX8: 0, RX9: 0, RX10: 0, RX11: 0, RX12: 2047, RX13: 0, RX14: 0, RX15: 15}
  DEBUG    root:machine.py:335 {TICK: 69, RX1: 7, RX2: 0, RX3: 0, RX4: 114, RX5: 112, RX6: 0, RX7: 0, RX8: 0, RX9: 0, RX10: 0, RX11: 0, RX12: 2047, RX13: 0, RX14: 0, RX15: 15}
  DEBUG    root:machine.py:335 {TICK: 73, RX1: 8, RX2: 0, RX3: 0, RX4: 114, RX5: 112, RX6: 0, RX7: 0, RX8: 0, RX9: 0, RX10: 0, RX11: 0, RX12: 2047, RX13: 0, RX14: 0, RX15: 15}
  DEBUG    root:machine.py:335 {TICK: 77, RX1: 9, RX2: 0, RX3: 0, RX4: 114, RX5: 112, RX6: 0, RX7: 0, RX8: 0, RX9: 0, RX10: 0, RX11: 0, RX12: 2047, RX13: 0, RX14: 0, RX15: 15}
  DEBUG    root:machine.py:335 {TICK: 81, RX1: 10, RX2: 0, RX3: 0, RX4: 114, RX5: 114, RX6: 0, RX7: 0, RX8: 0, RX9: 0, RX10: 0, RX11: 0, RX12: 2047, RX13: 0, RX14: 0, RX15: 15}
  INFO     root:machine.py:57 output: ['p'] << 'r'
  DEBUG    root:machine.py:335 {TICK: 84, RX1: 11, RX2: 0, RX3: 0, RX4: 114, RX5: 114, RX6: 0, RX7: 0, RX8: 0, RX9: 0, RX10: 0, RX11: 0, RX12: 2047, RX13: 0, RX14: 0, RX15: 15}
  DEBUG    root:machine.py:335 {TICK: 88, RX1: 12, RX2: 0, RX3: 0, RX4: 114, RX5: 114, RX6: 0, RX7: 0, RX8: 0, RX9: 0, RX10: 0, RX11: 0, RX12: 2047, RX13: 0, RX14: 0, RX15: 15}
  DEBUG    root:machine.py:335 {TICK: 92, RX1: 13, RX2: 1, RX3: 0, RX4: 114, RX5: 114, RX6: 0, RX7: 0, RX8: 0, RX9: 0, RX10: 0, RX11: 0, RX12: 2047, RX13: 0, RX14: 0, RX15: 15}
  DEBUG    root:machine.py:335 {TICK: 96, RX1: 14, RX2: 1, RX3: 0, RX4: 114, RX5: 114, RX6: 0, RX7: 0, RX8: 0, RX9: 0, RX10: 0, RX11: 0, RX12: 2047, RX13: 0, RX14: 0, RX15: 4}
  DEBUG    root:machine.py:335 {TICK: 98, RX1: 4, RX2: 1, RX3: 0, RX4: 114, RX5: 114, RX6: 0, RX7: 0, RX8: 0, RX9: 0, RX10: 0, RX11: 0, RX12: 2047, RX13: 0, RX14: 0, RX15: 4}
  DEBUG    root:machine.py:335 {TICK: 102, RX1: 5, RX2: 1, RX3: 0, RX4: 114, RX5: 114, RX6: 0, RX7: 0, RX8: 0, RX9: 0, RX10: 0, RX11: 0, RX12: 2047, RX13: 0, RX14: 0, RX15: 15}
  DEBUG    root:machine.py:335 {TICK: 106, RX1: 6, RX2: 0, RX3: 0, RX4: 114, RX5: 114, RX6: 0, RX7: 0, RX8: 0, RX9: 0, RX10: 0, RX11: 0, RX12: 2047, RX13: 0, RX14: 0, RX15: 15}
  DEBUG    root:machine.py:335 {TICK: 110, RX1: 7, RX2: 0, RX3: 0, RX4: 105, RX5: 114, RX6: 0, RX7: 0, RX8: 0, RX9: 0, RX10: 0, RX11: 0, RX12: 2047, RX13: 0, RX14: 0, RX15: 15}
  DEBUG    root:machine.py:335 {TICK: 114, RX1: 8, RX2: 0, RX3: 0, RX4: 105, RX5: 114, RX6: 0, RX7: 0, RX8: 0, RX9: 0, RX10: 0, RX11: 0, RX12: 2047, RX13: 0, RX14: 0, RX15: 15}
  DEBUG    root:machine.py:335 {TICK: 118, RX1: 9, RX2: 0, RX3: 0, RX4: 105, RX5: 114, RX6: 0, RX7: 0, RX8: 0, RX9: 0, RX10: 0, RX11: 0, RX12: 2047, RX13: 0, RX14: 0, RX15: 15}
  DEBUG    root:machine.py:335 {TICK: 122, RX1: 10, RX2: 0, RX3: 0, RX4: 105, RX5: 105, RX6: 0, RX7: 0, RX8: 0, RX9: 0, RX10: 0, RX11: 0, RX12: 2047, RX13: 0, RX14: 0, RX15: 15}
  INFO     root:machine.py:57 output: ['p', 'r'] << 'i'
  DEBUG    root:machine.py:335 {TICK: 125, RX1: 11, RX2: 0, RX3: 0, RX4: 105, RX5: 105, RX6: 0, RX7: 0, RX8: 0, RX9: 0, RX10: 0, RX11: 0, RX12: 2047, RX13: 0, RX14: 0, RX15: 15}
  DEBUG    root:machine.py:335 {TICK: 129, RX1: 12, RX2: 0, RX3: 0, RX4: 105, RX5: 105, RX6: 0, RX7: 0, RX8: 0, RX9: 0, RX10: 0, RX11: 0, RX12: 2047, RX13: 0, RX14: 0, RX15: 15}
  DEBUG    root:machine.py:335 {TICK: 133, RX1: 13, RX2: 1, RX3: 0, RX4: 105, RX5: 105, RX6: 0, RX7: 0, RX8: 0, RX9: 0, RX10: 0, RX11: 0, RX12: 2047, RX13: 0, RX14: 0, RX15: 15}
  DEBUG    root:machine.py:335 {TICK: 137, RX1: 14, RX2: 1, RX3: 0, RX4: 105, RX5: 105, RX6: 0, RX7: 0, RX8: 0, RX9: 0, RX10: 0, RX11: 0, RX12: 2047, RX13: 0, RX14: 0, RX15: 4}
  DEBUG    root:machine.py:335 {TICK: 139, RX1: 4, RX2: 1, RX3: 0, RX4: 105, RX5: 105, RX6: 0, RX7: 0, RX8: 0, RX9: 0, RX10: 0, RX11: 0, RX12: 2047, RX13: 0, RX14: 0, RX15: 4}
  DEBUG    root:machine.py:335 {TICK: 143, RX1: 5, RX2: 1, RX3: 0, RX4: 105, RX5: 105, RX6: 0, RX7: 0, RX8: 0, RX9: 0, RX10: 0, RX11: 0, RX12: 2047, RX13: 0, RX14: 0, RX15: 15}
  DEBUG    root:machine.py:335 {TICK: 147, RX1: 6, RX2: 0, RX3: 0, RX4: 105, RX5: 105, RX6: 0, RX7: 0, RX8: 0, RX9: 0, RX10: 0, RX11: 0, RX12: 2047, RX13: 0, RX14: 0, RX15: 15}
  DEBUG    root:machine.py:335 {TICK: 151, RX1: 7, RX2: 0, RX3: 0, RX4: 118, RX5: 105, RX6: 0, RX7: 0, RX8: 0, RX9: 0, RX10: 0, RX11: 0, RX12: 2047, RX13: 0, RX14: 0, RX15: 15}
  DEBUG    root:machine.py:335 {TICK: 155, RX1: 8, RX2: 0, RX3: 0, RX4: 118, RX5: 105, RX6: 0, RX7: 0, RX8: 0, RX9: 0, RX10: 0, RX11: 0, RX12: 2047, RX13: 0, RX14: 0, RX15: 15}
  DEBUG    root:machine.py:335 {TICK: 159, RX1: 9, RX2: 0, RX3: 0, RX4: 118, RX5: 105, RX6: 0, RX7: 0, RX8: 0, RX9: 0, RX10: 0, RX11: 0, RX12: 2047, RX13: 0, RX14: 0, RX15: 15}
  DEBUG    root:machine.py:335 {TICK: 163, RX1: 10, RX2: 0, RX3: 0, RX4: 118, RX5: 118, RX6: 0, RX7: 0, RX8: 0, RX9: 0, RX10: 0, RX11: 0, RX12: 2047, RX13: 0, RX14: 0, RX15: 15}
  INFO     root:machine.py:57 output: ['p', 'r', 'i'] << 'v'
  DEBUG    root:machine.py:335 {TICK: 166, RX1: 11, RX2: 0, RX3: 0, RX4: 118, RX5: 118, RX6: 0, RX7: 0, RX8: 0, RX9: 0, RX10: 0, RX11: 0, RX12: 2047, RX13: 0, RX14: 0, RX15: 15}
  DEBUG    root:machine.py:335 {TICK: 170, RX1: 12, RX2: 0, RX3: 0, RX4: 118, RX5: 118, RX6: 0, RX7: 0, RX8: 0, RX9: 0, RX10: 0, RX11: 0, RX12: 2047, RX13: 0, RX14: 0, RX15: 15}
  DEBUG    root:machine.py:335 {TICK: 174, RX1: 13, RX2: 1, RX3: 0, RX4: 118, RX5: 118, RX6: 0, RX7: 0, RX8: 0, RX9: 0, RX10: 0, RX11: 0, RX12: 2047, RX13: 0, RX14: 0, RX15: 15}
  DEBUG    root:machine.py:335 {TICK: 178, RX1: 14, RX2: 1, RX3: 0, RX4: 118, RX5: 118, RX6: 0, RX7: 0, RX8: 0, RX9: 0, RX10: 0, RX11: 0, RX12: 2047, RX13: 0, RX14: 0, RX15: 4}
  DEBUG    root:machine.py:335 {TICK: 180, RX1: 4, RX2: 1, RX3: 0, RX4: 118, RX5: 118, RX6: 0, RX7: 0, RX8: 0, RX9: 0, RX10: 0, RX11: 0, RX12: 2047, RX13: 0, RX14: 0, RX15: 4}
  DEBUG    root:machine.py:335 {TICK: 184, RX1: 5, RX2: 1, RX3: 0, RX4: 118, RX5: 118, RX6: 0, RX7: 0, RX8: 0, RX9: 0, RX10: 0, RX11: 0, RX12: 2047, RX13: 0, RX14: 0, RX15: 15}
  DEBUG    root:machine.py:335 {TICK: 188, RX1: 6, RX2: 0, RX3: 0, RX4: 118, RX5: 118, RX6: 0, RX7: 0, RX8: 0, RX9: 0, RX10: 0, RX11: 0, RX12: 2047, RX13: 0, RX14: 0, RX15: 15}
  DEBUG    root:machine.py:335 {TICK: 192, RX1: 7, RX2: 0, RX3: 0, RX4: 101, RX5: 118, RX6: 0, RX7: 0, RX8: 0, RX9: 0, RX10: 0, RX11: 0, RX12: 2047, RX13: 0, RX14: 0, RX15: 15}
  DEBUG    root:machine.py:335 {TICK: 196, RX1: 8, RX2: 0, RX3: 0, RX4: 101, RX5: 118, RX6: 0, RX7: 0, RX8: 0, RX9: 0, RX10: 0, RX11: 0, RX12: 2047, RX13: 0, RX14: 0, RX15: 15}
  DEBUG    root:machine.py:335 {TICK: 200, RX1: 9, RX2: 0, RX3: 0, RX4: 101, RX5: 118, RX6: 0, RX7: 0, RX8: 0, RX9: 0, RX10: 0, RX11: 0, RX12: 2047, RX13: 0, RX14: 0, RX15: 15}
  DEBUG    root:machine.py:335 {TICK: 204, RX1: 10, RX2: 0, RX3: 0, RX4: 101, RX5: 101, RX6: 0, RX7: 0, RX8: 0, RX9: 0, RX10: 0, RX11: 0, RX12: 2047, RX13: 0, RX14: 0, RX15: 15}
  INFO     root:machine.py:57 output: ['p', 'r', 'i', 'v'] << 'e'
  DEBUG    root:machine.py:335 {TICK: 207, RX1: 11, RX2: 0, RX3: 0, RX4: 101, RX5: 101, RX6: 0, RX7: 0, RX8: 0, RX9: 0, RX10: 0, RX11: 0, RX12: 2047, RX13: 0, RX14: 0, RX15: 15}
  DEBUG    root:machine.py:335 {TICK: 211, RX1: 12, RX2: 0, RX3: 0, RX4: 101, RX5: 101, RX6: 0, RX7: 0, RX8: 0, RX9: 0, RX10: 0, RX11: 0, RX12: 2047, RX13: 0, RX14: 0, RX15: 15}
  DEBUG    root:machine.py:335 {TICK: 215, RX1: 13, RX2: 1, RX3: 0, RX4: 101, RX5: 101, RX6: 0, RX7: 0, RX8: 0, RX9: 0, RX10: 0, RX11: 0, RX12: 2047, RX13: 0, RX14: 0, RX15: 15}
  DEBUG    root:machine.py:335 {TICK: 219, RX1: 14, RX2: 1, RX3: 0, RX4: 101, RX5: 101, RX6: 0, RX7: 0, RX8: 0, RX9: 0, RX10: 0, RX11: 0, RX12: 2047, RX13: 0, RX14: 0, RX15: 4}
  DEBUG    root:machine.py:335 {TICK: 221, RX1: 4, RX2: 1, RX3: 0, RX4: 101, RX5: 101, RX6: 0, RX7: 0, RX8: 0, RX9: 0, RX10: 0, RX11: 0, RX12: 2047, RX13: 0, RX14: 0, RX15: 4}
  DEBUG    root:machine.py:335 {TICK: 225, RX1: 5, RX2: 1, RX3: 0, RX4: 101, RX5: 101, RX6: 0, RX7: 0, RX8: 0, RX9: 0, RX10: 0, RX11: 0, RX12: 2047, RX13: 0, RX14: 0, RX15: 15}
  DEBUG    root:machine.py:335 {TICK: 229, RX1: 6, RX2: 0, RX3: 0, RX4: 101, RX5: 101, RX6: 0, RX7: 0, RX8: 0, RX9: 0, RX10: 0, RX11: 0, RX12: 2047, RX13: 0, RX14: 0, RX15: 15}
  DEBUG    root:machine.py:335 {TICK: 233, RX1: 7, RX2: 0, RX3: 0, RX4: 116, RX5: 101, RX6: 0, RX7: 0, RX8: 0, RX9: 0, RX10: 0, RX11: 0, RX12: 2047, RX13: 0, RX14: 0, RX15: 15}
  DEBUG    root:machine.py:335 {TICK: 237, RX1: 8, RX2: 0, RX3: 0, RX4: 116, RX5: 101, RX6: 0, RX7: 0, RX8: 0, RX9: 0, RX10: 0, RX11: 0, RX12: 2047, RX13: 0, RX14: 0, RX15: 15}
  DEBUG    root:machine.py:335 {TICK: 241, RX1: 9, RX2: 0, RX3: 0, RX4: 116, RX5: 101, RX6: 0, RX7: 0, RX8: 0, RX9: 0, RX10: 0, RX11: 0, RX12: 2047, RX13: 0, RX14: 0, RX15: 15}
  DEBUG    root:machine.py:335 {TICK: 245, RX1: 10, RX2: 0, RX3: 0, RX4: 116, RX5: 116, RX6: 0, RX7: 0, RX8: 0, RX9: 0, RX10: 0, RX11: 0, RX12: 2047, RX13: 0, RX14: 0, RX15: 15}
  INFO     root:machine.py:57 output: ['p', 'r', 'i', 'v', 'e'] << 't'
  DEBUG    root:machine.py:335 {TICK: 248, RX1: 11, RX2: 0, RX3: 0, RX4: 116, RX5: 116, RX6: 0, RX7: 0, RX8: 0, RX9: 0, RX10: 0, RX11: 0, RX12: 2047, RX13: 0, RX14: 0, RX15: 15}
  DEBUG    root:machine.py:335 {TICK: 252, RX1: 12, RX2: 0, RX3: 0, RX4: 116, RX5: 116, RX6: 0, RX7: 0, RX8: 0, RX9: 0, RX10: 0, RX11: 0, RX12: 2047, RX13: 0, RX14: 0, RX15: 15}
  INFO     root:machine.py:337 Input buffer is empty!
  INFO     root:machine.py:361 output:  ['p', 'r', 'i', 'v', 'e', 't'], ticks: 252

output: |
  Output buffer: privet | ticks: 252 | amount_instr: 67